### PR TITLE
Scrum 5 create model form

### PR DESCRIPTION
**Pull request by Umarfarook1912**
### Things I had implemented in my task

_________________

#### PR Checklist
- [x] Develop a UI design for Model API form.
- [x] Implement hooks form for easy validation.
- [x] Display error message with help of formstate.
- [x] ToastContainer for display toast message.

### CODE REVIEW

File: src/pages/model-form/model-form/ModelForm.css
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.css:1]: Rename folder name to caps ModelForm and dont create multiple subfolders with same name modle-form/model-form

File: src/pages/model-form/model-form/ModelForm.css [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    text-align: center;
+}
+
+
+.input-row {
+    display: flex;
+    justify-content: space-between;
+    align-items: flex-start;
+    gap: 30px;
+    width: 1230px;
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.css:38]: Dont set width explicitly

File: src/pages/model-form/model-form/ModelForm.css [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+.label-row label {
+    flex: 1;
+    text-align: center;
+}
+
+
+.input-row {
+    display: flex;
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.css:34]: try using **d-flex justify-content-between align-start** in className

File: src/pages/model-form/model-form/ModelForm.css [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    justify-content: space-between;
+    align-items: flex-start;
+    gap: 30px;
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.css:37]: use gap-4

File: src/pages/model-form/model-form/ModelForm.css [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    padding: 20px;
+    border-radius: 8px;
+    background-color: #f9f9f9;
+    margin-top: 20px;
+    width:1310px;
+}
+
+
+.label-row {
+    display: flex;
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.css:16]: same use **d-flex** classname

File: src/App.tsx [hunk 5]
```diff
@@ -1,24 +1,21 @@
 import logo from './logo.svg';
 import './App.css';
+import 'bootstrap/dist/css/bootstrap.min.css';
```
Comment (mentorbridgeindia) [inline src/App.tsx:3]: import this in App.css, not here

File: src/pages/model-form/model-form/ModelForm.css [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+.d-flex button:last-child:hover {
+    background-color: #5a6268;
+}
+
+label.Null {
+    padding-left: 100px;
+}
+
+.submit-cancel-buttons {
+    width: 86px;
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.css:126]: Don't set width explicitly

File: src/pages/model-form/model-form/ModelForm.css [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    height: 45px;
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.css:127]: Don't set height explicitly

File: src/pages/model-form/model-form/ModelForm.css [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+    margin-left: -30px;
+    
+  }
+
+  .model-name-label{
+     margin-left: 48px;
+     font-weight: bold;
+     padding-top: 6px;
+     padding-left: 35px;
+     font-family: 'Times New Roman', Times, serif;
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.css:149]: Font name changed for a particular element? why?

File: src/pages/model-form/model-form/ModelForm.css [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+    padding-bottom: 2px;
+  }
+  .error-message{
+    color: red;
+  }
+  .my-4{
+    font-size: 40px;
+    font-weight: bold;
+    text-align: center;
+    font-family: 'Times New Roman', Times, serif;
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.css:175]: Again font family changed

File: src/pages/model-form/model-form/ModelForm.tsx [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+    name: "variables",
+  });
+
+  const onSubmit = (data: ModelData) => {
+    const jsonData = {
+      modelName: data.modelName,
+      variables: data.variables,
+    };
+
+    console.log("Generated JSON:", jsonData);
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.tsx:34]: Remove console logs before committing

File: src/pages/model-form/model-form/ModelForm.tsx [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+    toast.info("Form reset");
+  };
+
+  return (
+    <Container>
+      <ToastContainer position="top-center" />
+      <h1 className="my-4 text-center">CREATE API MODELS</h1>
+      <div className="model-container">
+        <Form onSubmit={handleSubmit(onSubmit)}>
+          {/* Model Name Input */}
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.tsx:55]: Remove unwanted commented lines

File: src/pages/model-form/model-form/ModelForm.tsx [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+                type="text"
+                placeholder="Enter model name"
+                className="model-name-input"
+              />
+              {errors?.modelName && <p className="error-message">{errors.modelName.message}</p>}
+            </Col>
+          </Row>
+          <hr></hr>
+
+          {/* Variables Section */}
```
Comment (mentorbridgeindia) [inline src/pages/model-form/model-form/ModelForm.tsx:72]: remove comment

**Review by mentorbridgeindia — CHANGES_REQUESTED** [review]

**Review by mentorbridgeindia — APPROVED** [review]

**Comment by mentorbridgeindia** [comment]
Resolve conflicts before you merge

**Review by mentorbridgeindia — CHANGES_REQUESTED** [review]
Resolve merge conflicts

**Comment by mentorbridgeindia** [comment]
Take latest pull from main and resolve conflicts

**Comment by Umarfarook1912** [comment]
I solve the conflicts and merge my branch sir thank you .Please proceed further

**Review by Umarfarook1912 — COMMENTED** [review]
completed

**Pull request closed without merging**